test_cases/test_dojo.py

```diff
@@ -97,23 +100,136 @@ def test_print_unallocated_with_filename_unspecified(self):
-		self.the_dojo.print_unallocated("file2")
-		self.assertTrue(os.path.isfile("file2.txt"))
-		os.remove("file2.txt")
+	def test_reallocate_person_returns_error_if_room_doesnot_exist(self):
+		self.the_dojo.create_room("office", "Red")
+		self.the_dojo.add_person("Pete", "pete@pete", "staff")
+		result =self.the_dojo.reallocate_person("pete@pete", "Green")
+		self.assertEqual(result, "The OFFICE Green doesnot exist!")
 
+	def test_reallocate_person_successfully_reallocates_person(self):
+		self.the_dojo.create_room("office", "Red")
+		self.the_dojo.add_person("Pete", "pete@pete", "staff")
+		self.the_dojo.create_room("office", "Green")
+		room_occupancy = [room.occupants for room in self.the_dojo.all_offices\
+		 if room.name == "Green"]
+		count_before_reallocation = len(room_occupancy[0])
+		self.the_dojo.reallocate_person("pete@pete", "Green")
+		room_occupancy = [room.occupants for room in self.the_dojo.all_offices\
+		 if room.name == "Green"]
+		count_after_reallocation = len(room_occupancy[0])
+		self.assertEqual(count_after_reallocation - count_before_reallocation, 1, 
+			msg = "Reallocation was unsuccessful!" )
+
+	def test_reallocate_person_moves_person_from_previous_allocation(self):
+		self.the_dojo.create_room("office", "Red")
+		self.the_dojo.add_person("Pete", "pete@pete", "staff")
+		self.the_dojo.create_room("office", "Green")
+		room_occupancy = [room.occupants for room in self.the_dojo.all_offices\
+		 if room.name == "Red"]
+		count_before_reallocation = len(room_occupancy[0])
+		self.the_dojo.reallocate_person("pete@pete", "Green")
+		room_occupancy = [room.occupants for room in self.the_dojo.all_offices\
+		 if room.name == "Red"]
+		count_after_reallocation = len(room_occupancy[0])
+		self.assertEqual(count_before_reallocation - count_after_reallocation, 1, 
+			msg = "Reallocation was unsuccessful!" )
+		
+	def test_reallocate_person_doesnot_reallocate_staff_to_livingspace(self):
+		self.the_dojo.create_room("office", "Red")
+		self.the_dojo.add_person("Pete", "pete@pete", "staff")
+		self.the_dojo.create_room("livingspace", "Crimson")
+		result = self.the_dojo.reallocate_person("pete@pete", "Crimson")
+		self.assertEqual(result, "Cannot reallocate STAFF to LIVING SPACE!")
+
+	def test_reallocate_person_doesnot_reallocate_if_room_is_full(self):
+		self.the_dojo.create_room("office", "Red")
+		self.the_dojo.add_person("Pete", "pete@pete", "staff")
+		self.the_dojo.add_person("Matt", "matt@matt", "staff")
+		self.the_dojo.add_person("Patt", "patt@patt", "fellow")
+		self.the_dojo.add_person("Faith", "faith@faith", "staff")
+		self.the_dojo.add_person("Fait", "fait@fait", "fellow", "y")
+		self.the_dojo.add_person("Fai", "fai@fai", "staff")
+		self.the_dojo.create_room("office", "Green")
+		self.the_dojo.add_person("Marie", "maz@maz", "fellow","y")
+		result = self.the_dojo.reallocate_person("maz@maz", "Red")
+		self.assertEqual(result, "The room Red is full! Cannot reallocate FELLOW-Marie")
```

Comment:
Once you implement the functionality, you can reduce the visual noise by using the load_people function to load the members